core/src/main/scala/org/apache/spark/deploy/security/ServiceCredentialProvider.scala

```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package org.apache.spark.deploy.yarn.security
+package org.apache.spark.deploy.security
 
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.security.{Credentials, UserGroupInformation}
```

Comment:
OK, I:

- Readded `deploy.yarn.security.ServiceCredentialProvider`, extended from `deploy.security.ServiceCredentialProvider`, and deprecated it.
- Added a new `YARNConfigurableCredentialProvider`, which extends from `ConfigurableCredentialProvider`, with the added behavior that it loads providers from `deploy.yarn.security.ServiceCredentialProvider` as well.